linera-views/src/dynamo_db.rs

```diff
@@ -887,9 +823,208 @@ impl DynamoDbClientInternal {
     }
 }
 
+struct QueryResults {
+    prefix_len: usize,
+    responses: Vec<QueryOutput>,
+}
+
+// Inspired by https://depth-first.com/articles/2020/06/22/returning-rust-iterators/
+#[doc(hidden)]
+#[allow(clippy::type_complexity)]
+pub struct DynamoDbKeyBlockIterator<'a> {
+    prefix_len: usize,
+    pos: usize,
+    iters: Vec<
```

Comment:
I tried to do that for a day and it was quite intricate and I failed.
If that is okay, I would ask for a bailout or help. Yes it would be nicer but that is difficult.

Reply:
sure no problem